PhysicsTools/IsolationAlgos/plugins/CITKPFIsolationSumProducerForPUPPI.cc

```diff
@@ -0,0 +1,182 @@
+#ifndef PhysicsTools_IsolationAlgos_CITKIsolationSumProducer_H
+#define PhysicsTools_IsolationAlgos_CITKIsolationSumProducer_H
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+#include "DataFormats/RecoCandidate/interface/IsoDepositFwd.h"
+#include "DataFormats/RecoCandidate/interface/IsoDeposit.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "PhysicsTools/IsolationAlgos/interface/EventDependentAbsVeto.h"
+
+#include "DataFormats/Candidate/interface/CandidateFwd.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "PhysicsTools/IsolationAlgos/interface/CITKIsolationConeDefinitionBase.h"
+#include "DataFormats/Common/interface/OwnVector.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include <string>
+#include <unordered_map>
+
+namespace edm { class Event; }
+namespace edm { class EventSetup; }
+
+namespace citk {
+  class PFIsolationSumProducerForPUPPI : public edm::EDProducer {
```

Comment:
all new modules should be thread-aware
This one looks like it can be edm::global